staging/src/k8s.io/cli-runtime/pkg/resource/helper.go

```diff
@@ -203,6 +203,24 @@ func (m *Helper) DeleteWithOptions(namespace, name string, options *metav1.Delet
 		Get()
 }
 
+// DeleteWithOptionsWithContext performs a DELETE request with the given context for timeout control
+func (m *Helper) DeleteWithOptionsWithContext(ctx context.Context, namespace, name string, options *metav1.DeleteOptions) (runtime.Object, error) {
```

Comment:
This duplicates `DeleteWithOptions`. I think it would be better to let `DeleteWithOptions` call the new `DeleteWithOptionsWithContext`:

```
func (m *Helper) DeleteWithOptions(namespace, name string, options *metav1.DeleteOptions) (runtime.Object, error) {
    return m.DeleteWithOptionsWithContext(context.Background(), name, options)
}
```

That gets rid of one `context.TODO`.